Source/EasyNetQ/RabbitBus.cs

```diff
@@ -62,25 +69,41 @@ public class RabbitBus : IBus
         {
             Preconditions.CheckNotNull(message, "message");
 
-            return PublishAsync(message, conventions.TopicNamingConvention(typeof(T)));
+            return PublishAsync(message, x => { });
         }
 
         public virtual Task PublishAsync<T>(T message, string topic) where T : class
         {
             Preconditions.CheckNotNull(message, "message");
             Preconditions.CheckNotNull(topic, "topic");
-            var messageType = typeof (T);
+
+            return PublishAsync(message, x => x.WithTopic(topic));
+        }
+
+        public Task PublishAsync<T>(T message, Action<IPublishConfiguration> configure) where T : class
+        {
+            Preconditions.CheckNotNull(message, "message");
+            Preconditions.CheckNotNull(configure, "configure");
+
+            var configuration = new PublishConfiguration(conventions.TopicNamingConvention(typeof(T)));
+            configure(configuration);
+
+            var messageType = typeof(T);
             return publishExchangeDeclareStrategy.DeclareExchangeAsync(advancedBus, messageType, ExchangeType.Topic).Then(exchange =>
+            {
+                var easyNetQMessage = new Message<T>(message)
                 {
-                    var easyNetQMessage = new Message<T>(message)
-                    {
-                        Properties =
+                    Properties =
                         {
                             DeliveryMode = messageDeliveryModeStrategy.GetDeliveryMode(messageType)
                         }
-                    };
-                    return advancedBus.PublishAsync(exchange, topic, false, false, easyNetQMessage); 
-                });
+                };
+                if (configuration.Priority != null)
+                    easyNetQMessage.Properties.Priority = configuration.Priority.Value;
+                if (configuration.Expires != null)
+                    easyNetQMessage.Properties.Expiration = configuration.Expires.ToString();
```

Comment:
[MessageProperties](https://github.com/EasyNetQ/EasyNetQ/blob/master/Source/EasyNetQ/MessageProperties.cs#L58) has flags `somePropertyPresent`.  These `if`s prevents from setting the flags to `true`.  Is these checks necessary or they could be removed?